pandas/tests/extension/base/setitem.py

```diff
@@ -195,3 +195,27 @@ def test_setitem_preserves_views(self, data):
         data[0] = data[1]
         assert view1[0] == data[1]
         assert view2[0] == data[1]
+
+    def test_setitem_nullable_integer(self, data):
```

Comment:
These base tests shouldn't make any assumptions about the data or dtype. They should only assume / assert things that are true for all extension arrays. So something like

```python
data = data[:5]
expected = data.take([0, 0, 0, 3, 4])
mask = pd.array([True, True, True, False, False])
data[mask] = data[0]
self.assert_array_equal(result, expected)
```

Reply:
ahh! I see, Thanks @TomAugspurger !

May I ask a question which is not very related: how are these tests tested? seem those test files will be skipped? 🤔

Reply:
https://dev.pandas.io/docs/development/extending.html#testing-extension-arrays.

So all the files in `pandas/tests/extension` inherit from the base classes.

Reply:
ahh! thanks for enlightening!! appreciate a lot! 👍